FlowCrypt/Controllers/Threads/ThreadDetailsViewController.swift

```diff
@@ -308,6 +310,43 @@ extension ThreadDetailsViewController {
             updateSpinner(label: "decrypting_title".localized)
         }
     }
+
+    private func handleKeyMismatch(for rawMimeData: Data, at indexPath: IndexPath) {
+        let alert = UIAlertController(
+            title: "error_key_mismatch".localized,
+            message: nil,
+            preferredStyle: .alert
+        )
+
+        alert.addAction(
+            UIAlertAction(
+                title: "go_back".localized,
+                style: .cancel,
+                handler: { [weak self] _ in
+                    self?.navigationController?.popViewController(animated: true)
+                }
+            )
+        )
+
+        alert.addAction(
+            UIAlertAction(
+                title: "message_open_anyway".localized,
+                style: .default,
+                handler: { [weak self] _ in
+                    let processedMessage = ProcessedMessage(
+                        rawMimeData: rawMimeData,
+                        text: String(data: rawMimeData, encoding: .utf8) ?? "",
+                        attachments: [],
+                        messageType: .encrypted,
+                        signature: .unsigned
```

Comment:
@tomholub
I decided to use `unsigned` for such kind of message. Are you OK with it?

Reply:
`messageType` should have another type that says `.error(decryptError)` where `decryptError` would be one of `keyMismatch`, `noMdc`, `badMdc` etc as returned by TypeScript. Then it should render red color: `error: key mismatch` or `error: bad mdc` etc. Let's do that for all error types.

In these situations - ts returned error - signature badge should be hidden. So I would just add a type `signture: .hidden` that doesn't render it.